File: combinatorics/resource/ST/project_4/bai_toan_3/Recursive.py
def p_self_recursive(n, k):
    if n == 0 and k == 0:
        return 1
    if n < 0 or k <= 0:
        return 0
    return p_self_recursive(n - (2 * k - 1), k - 1)

# Example usage
n, k = 13, 3
print(f"p_{{{k}}}^{{selfcjg}}({n}) =", p_self_recursive(n, k))

File: combinatorics/resource/ST/project_4/bai_toan_3/Recursive.cpp
#include <iostream>
using namespace std;

int p_self_recursive(int n, int k) {
    if (n == 0 && k == 0) return 1;
    if (n < 0 || k <= 0) return 0;
    return p_self_recursive(n - (2 * k - 1), k - 1);
}

int main() {
    int n = 13, k = 3;
    cout << "p_" << k << "^{selfcjg}(" << n << ") = "
         << p_self_recursive(n, k) << endl;
    return 0;
}